lambdas/services/base/s3_service.py

```diff
@@ -12,41 +13,49 @@
 class S3Service:
     _instance = None
 
-    def __new__(cls):
+    def __new__(cls, *args, **kwargs):
         if cls._instance is None:
             cls._instance = super().__new__(cls)
             cls._instance.initialised = False
         return cls._instance
 
-    def __init__(self):
+    def __init__(self, custom_aws_role=None):
         if not self.initialised:
             config = BotoConfig(
                 retries={"max_attempts": 3, "mode": "standard"},
                 s3={"addressing_style": "virtual"},
                 signature_version="s3v4",
             )
-            self.client = boto3.client("s3", config=config)
             self.presigned_url_expiry = 1800
+            self.client = boto3.client("s3", config=config)
             self.initialised = True
+            self.custom_client = None
+            if custom_aws_role:
+                iam_service = IAMService()
+                self.custom_client = iam_service.assume_role(
+                    custom_aws_role, "s3", config=config
+                )
 
     # S3 Location should be a minimum of a s3_object_key but can also be a directory location in the form of
```

Comment:
Wonder if we still need this comment? I know it's not part of your work but seems a bit unnecessary to me?

Reply:
I think we do. It's not clear without it.